lib/rubygems.rb

```diff
@@ -1270,6 +1270,14 @@ def register_default_spec(spec)
           next unless $~
         end
 
+        expanded_candidates = spec.full_require_paths.map do |path|
```

Comment:
I just noticed that this outer loop is unnecessary for gems with "new format gemspecs", and that my patch this doesn't properly fix the issue for gems with "old format gemspecs". I'll work on fixing this.